reactor-pool-micrometer/src/main/java/reactor/pool/introspection/micrometer/MicrometerMetricsRecorder.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2022 VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.pool.introspection.micrometer;
+
+import java.util.concurrent.TimeUnit;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+
+import reactor.pool.PoolMetricsRecorder;
+
+final class MicrometerMetricsRecorder implements PoolMetricsRecorder {
+
+	private final String        metricsPrefix;
+	private final MeterRegistry meterRegistry;
+
+	private final Timer               allocationFailureTimer;
+	private final Timer               allocationSuccessTimer;
+	private final DistributionSummary destroyedMeter;
+	private final Counter             recycledCounter;
+	private final Counter             recycledNotableFastPathCounter;
+	private final Counter             recycledNotableSlowPathCounter;
+	private final DistributionSummary resetMeter;
+	private final DistributionSummary resourceSummaryIdleness;
+	private final DistributionSummary resourceSummaryLifetime;
+
+	MicrometerMetricsRecorder(String metricsPrefix, MeterRegistry registry) {
+		this.metricsPrefix = metricsPrefix;
+		this.meterRegistry = registry;
+
+		allocationSuccessTimer = this.meterRegistry.timer(
+			DocumentedPoolMeters.ALLOCATION.getName(this.metricsPrefix),
+			Tags.of(DocumentedPoolMeters.AllocationTags.OUTCOME_SUCCESS));
```

Comment:
Static imports might help to make this nicer:
```java
allocationSuccessTimer = this.meterRegistry.timer(ALLOCATION.getName(this.metricsPrefix), OUTCOME_SUCCESS);
```
I'm also not sure if you need `Tags.of` here.

Reply:
why not :) the `Tags.of` is needed, as there is no signature that merely accepts a `Tag`